vignettes/tech-dm-keyed.Rmd

```diff
@@ -0,0 +1,216 @@
+enum_pk_candidates(weather_tbl)
+
+# Seems we have to construct a column with unique values
+# This can be done by combining column `origin` with `time_hour`, if the latter
+# is converted to a single time zone first; all within the `dm`:
+weather_tbl_mutate <-
+  weather_tbl %>%
+  # first convert all times to the same time zone:
+  mutate(time_hour_fmt = format(time_hour, tz = "UTC")) %>%
+  # paste together as character the airport code and the time
+  unite("origin_slot_id", origin, time_hour_fmt) %>%
+  select(origin_slot_id, everything())
+
+# check if we the result is as expected:
+enum_pk_candidates(weather_tbl_mutate) %>% filter(candidate)
+
+flights_upd_weather_dm <-
+  dm(
+    weather = weather_tbl_mutate,
+    flights_dm[c("airlines", "airports", "planes", "flights")]
+  ) %>%
+  dm_add_pk(weather, origin_slot_id)
+
+flights_upd_weather_dm
+
+# creating the coveted FK relation between `flights` and `weather`
+extended_flights_tbl <-
+  flights_upd_weather_dm$flights %>%
+  mutate(time_hour_fmt = format(time_hour, tz = "UTC")) %>%
+  # need to keep `origin` as FK to airports, so `remove = FALSE`
+  unite("origin_slot_id", origin, time_hour_fmt, remove = FALSE)
+
+extended_flights_dm <- dm(
+  flights = extended_flights_tbl,
+  flights_dm[c("airlines", "airports", "planes", "weather")]
+) %>%
+  dm_add_fk(flights, origin_slot_id, weather)
+
+extended_flights_dm %>% dm_draw()
+```
+
+### Use case 3: Disentangle `dm`
+
+If you look at the `dm` created by `dm_nycflights13(cycle = TRUE)`, you see that two columns of `flights` relate to one and the same table, `airports`.
+One column stands for the departure airport and the other for the arrival airport.
+
+```{r}
+dm_draw(dm_nycflights13(cycle = TRUE))
+```
+
+In such cases, it can be beneficial to "disentangle" the `dm` by duplicating the referred table.
+One way to do this in the {dm}-framework is as follows:
+
+```{r, error=TRUE}
+flights_cycle_dm <- dm_nycflights13(cycle = TRUE)
+
+disentangled_flights_dm <-
+  dm(
+    destination = flights_cycle_dm$airports,
+    origin = flights_cycle_dm$airports,
+    flights_dm[c("flights", "airlines", "planes", "weather")]
+  ) %>%
+  # Key relations are also duplicated, so the wrong ones need to be removed
+  dm_rm_fk(flights, dest, origin) %>%
+  dm_rm_fk(flights, origin, destination)
+
+dm_draw(disentangled_flights_dm)
+```
+
+In a future update, we will provide a more convenient way to "disentangle" `dm` objects, so that the individual steps will be done automatically.
+
+### Use case 4: Add summary table to `dm`
+
+Here is an example for adding a summary of a table as a new table to a `dm` (FK-relations are taken care of automatically):
+
+```{r, error=TRUE}
+dm_with_summary <-
+  flights_dm_tbl$flights %>%
+  count(origin, carrier) 
+
+flights_dm <- dm(
+  dep_carrier_count = dm_with_summary,
+  flights_dm[c("airlines", "airports", "flights", "planes", "weather")]
+)
+
+dm_draw(flights_dm)
+```
+
+### Use case 5: Joining tables
```

Comment:
why use that rather than flatten/squash? Maybe add the differences here?

Reply:
Too much detail, I'd say. This is about enabling users to keep their existing dplyr workflows.